source/common/network/lc_trie.h

```diff
@@ -445,21 +445,13 @@ template <class T> class LcTrie {
       ip_prefixes_ = data;
       std::sort(ip_prefixes_.begin(), ip_prefixes_.end());
 
-      // In theory, the trie_ vector can have at most twice the number of ip_prefixes entries - 1.
-      // However, due to the fill factor a buffer is added to the size of the
-      // trie_. The buffer value(2000000) is reused from the reference implementation in
-      // http://www.csc.kth.se/~snilsson/software/router/C/trie.c.
-      // TODO(ccaraman): Define a better buffer value when resizing the trie_.
-      maximum_trie_node_size = 2 * ip_prefixes_.size() + 2000000;
-      trie_.resize(maximum_trie_node_size);
-
       // Build the trie_.
+      trie_.reserve(size_t(ip_prefixes_.size() / fill_factor_));
```

Comment:
The `size_t` constructor looks bizarre, but I'm not C++ expert. @ccaraman / @mattklein123 could you take a look?

Reply:
It's fine, could also use a static cast.